tests/test_all.py

```diff
@@ -24,20 +24,16 @@ def test_questions_match_schema(all_files):
 
 
 @pytest.fixture
-def all_files():
-    all_paths = []
-    root_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
-    g6_root_dir = os.path.join(root_dir, 'g6')
-    g7d_root_dir = os.path.join(root_dir, 'g7_declaration')
-    for filename in os.listdir(g6_root_dir):
-        if filename.endswith('.yml') and filename != 'index.yml':
-            filepath = os.path.join(g6_root_dir, filename)
-            all_paths.append(filepath)
-    for filename in os.listdir(g7d_root_dir):
-        if filename.endswith('.yml') and filename != 'index.yml':
-            filepath = os.path.join(g7d_root_dir, filename)
-            all_paths.append(filepath)
-    return all_paths
+def get_all_files():
+
+    def all_files():
```

Comment:
Does it need to be a generator? If it does you could call the function in the outer function to return the generator rather than the function.